enterprise/mqtt/src/main/java/io/crate/mqtt/operations/MqttIngestService.java

```diff
@@ -168,24 +180,66 @@ public void doInsert(String clientId, MqttPublishMessage msg, BiConsumer<Object,
         List<Object> argsAsList = Arrays.asList(args);
 
         boolean messageMatchedRule = false;
+        boolean callbackNotified = false;
         SQLOperations.Session session = sqlOperations.createSession(Schemas.DOC_SCHEMA_NAME, crateUser, Option.NONE, 1);
+        List<CompletableFuture> insertOperationsFuture = new ArrayList<>(predicateAndIngestRules.size());
         for (Tuple<Predicate<Row>, IngestRule> entry : predicateAndIngestRules) {
             if (entry.v1().test(new RowN(args))) {
                 messageMatchedRule = true;
                 IngestRule ingestRule = entry.v2();
 
-                session.parse(STATEMENT_NAME, "insert into " + TableIdent.fromIndexName(ingestRule.getTargetTable()).fqn() +
-                                              " (\"client_id\", \"packet_id\", \"topic\", \"ts\", \"payload\") " +
-                                              "values (?, ?, ?, CURRENT_TIMESTAMP, ?)", FIELD_TYPES);
-                session.bind(SQLOperations.Session.UNNAMED, STATEMENT_NAME, argsAsList, null);
-                ResultReceiver resultReceiver = new BaseResultReceiver();
-                resultReceiver.completionFuture().whenComplete(ackCallback);
-                session.execute(SQLOperations.Session.UNNAMED, 0, resultReceiver);
-                session.sync();
+                try {
+                    session.parse(ingestRule.getName(), "insert into " + TableIdent.fromIndexName(ingestRule.getTargetTable()).fqn() +
+                                                  " (\"client_id\", \"packet_id\", \"topic\", \"ts\", \"payload\") " +
+                                                  "values (?, ?, ?, CURRENT_TIMESTAMP, ?)", FIELD_TYPES);
+                    session.bind(SQLOperations.Session.UNNAMED, ingestRule.getName(), argsAsList, null);
+                    BaseResultReceiver resultReceiver = new BaseResultReceiver();
+                    insertOperationsFuture.add(resultReceiver.completionFuture());
```

Comment:
Maybe check the `isDup` flag already here and swallow any VersionEngineExceptions here?

`insertOperationsFuture.add(resultReceiver.completionFuture().exceptionally(maybeSwallow))`

I think this would make the `allResultsComplete.whenComplete` logic further
below mostly unnecessary and it would also provide greater encapsulation (right
now the VersionConflictEngineException handling is in *two* places. First here,
and a second time in the `SendPubAckCallback`).  Ideally we'd only handle this
in one place.

It would probably also make sense to use `isDocumentAlreadyExistsException` -
given that `VersionConflictEngineException` is a bit ambiguous.